modules/tensor_mechanics/src/materials/ComputeElasticSmearedCrackingStress.C

```diff
@@ -0,0 +1,634 @@
+  params.addClassDescription("Compute stress using elasticity for finite strains");
+  params.addParam<std::string>("cracking_release", "abrupt", "The cracking release type.  Choices are abrupt (default) and exponential.");
+  params.addParam<Real>("cracking_stress", 0.0, "The stress threshold beyond which cracking occurs.  Must be positive.");
+  params.addParam<Real>("cracking_residual_stress", 0.0, "The fraction of the cracking stress allowed to be maintained following a crack.");
+  params.addParam<FunctionName>("cracking_stress_function", "", "The cracking stress as a function of time and location" );
+  params.addParam<std::vector<unsigned int> >("active_crack_planes", "Planes on which cracks are allowed (0,1,2 -> x,z,theta in RZ)");
+  params.addParam<unsigned int>("max_cracks", 3, "The maximum number of cracks allowed at a material point.");
+  params.addParam<Real>("cracking_neg_fraction", "The fraction of the cracking strain at which a transitition begins during decreasing strain to the original stiffness.");
+  return params;
+}
+
+namespace
+{
+  ComputeElasticSmearedCrackingStress::CRACKING_RELEASE
+  getCrackingModel( const std::string & name )
+  {
+    std::string n(name);
+    std::transform( n.begin(), n.end(), n.begin(), ::tolower);
+    ComputeElasticSmearedCrackingStress::CRACKING_RELEASE cm(ComputeElasticSmearedCrackingStress::CR_UNKNOWN);
+    if (n == "abrupt")
+      cm = ComputeElasticSmearedCrackingStress::CR_ABRUPT;
+    else if (n == "exponential")
+      cm = ComputeElasticSmearedCrackingStress::CR_EXPONENTIAL;
+    else if (n == "power")
+      cm = ComputeElasticSmearedCrackingStress::CR_POWER;
+    if (cm == ComputeElasticSmearedCrackingStress::CR_UNKNOWN)
+      mooseError("Unknown cracking model");
+    return cm;
+  }
+}
+
+ComputeElasticSmearedCrackingStress::ComputeElasticSmearedCrackingStress(const InputParameters & parameters) :
+    ComputeStressBase(parameters),
+    _mechanical_strain(getMaterialPropertyByName<RankTwoTensor>(_base_name + "mechanical_strain")),
+    _is_finite_strain(hasMaterialProperty<RankTwoTensor>(_base_name + "strain_increment")),
+    _strain_increment(getDefaultMaterialProperty<RankTwoTensor>(_base_name + "strain_increment")),
+    _rotation_increment(getDefaultMaterialProperty<RankTwoTensor>(_base_name + "rotation_increment")),
+    _stress_old(declarePropertyOld<RankTwoTensor>(_base_name + "stress")),
+    _elastic_strain_old(declarePropertyOld<RankTwoTensor>(_base_name + "elastic_strain")),
+    _cracking_release(getCrackingModel( getParam<std::string>("cracking_release"))),
+    _cracking_stress(parameters.isParamValid("cracking_stress") ?
+                    (getParam<Real>("cracking_stress") > 0 ? getParam<Real>("cracking_stress") : -1) : -1 ),
+    _cracking_residual_stress( getParam<Real>("cracking_residual_stress") ),
+    _cracking_stress_function( getParam<FunctionName>("cracking_stress_function") != "" ? &getFunction("cracking_stress_function") : NULL),
```

Comment:
Just declare `cracking_stress` as a function. This lets the user specify a plain number in the input file as well. It makes all this clutter unnecessary.